### ISSUE TITLE
Unstable target features are incorrectly reported for upstream crates

### ISSUE THREAD

**Issue by alexcrichton** [rust-lang/rust#50094]
Given:

```rust
// foo.rs
#![feature(mmx_target_feature)]

#[target_feature(enable = "mmx")]
pub unsafe fn foo() {}
```

and

```rust
extern crate foo;

fn main() {
    unsafe { foo::foo() }
}
```

you'll see:

```
$ rustc +nightly foo.rs --crate-type lib
$ rustc +nightly bar.rs -L .
error[E0658]: the target feature `mmx` is currently unstable
 --> /home/alex/code/rust4/foo.rs:3:18
  |
3 | #[target_feature(enable = "mmx")]
  |                  ^^^^^^^^^^^^^^
  |
  = help: add #![feature(mmx_target_feature)] to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
```

But that's not right! Only the original crate should need this directive, and it indeed has such a directive.
